src/halo-hooks/amm-v2/useSwap.ts

```diff
@@ -0,0 +1,228 @@
+import { Token } from '@halodao/sdk'
+// import { useEffect } from 'react'
+// import useBalancerSDK from './useBalancerSDK'
+import VaultABI from '../../constants/haloAbis/Vault.json'
+import { useActiveWeb3React } from 'hooks'
+import { formatUnits, parseUnits } from 'ethers/lib/utils'
+import useHaloAddresses from 'halo-hooks/useHaloAddresses'
+import { useContract } from 'hooks/useContract'
+import { consoleLog } from 'utils/simpleLogger'
+import assert from 'assert'
+import { /*BigNumber,*/ ethers } from 'ethers'
+import useToken from 'halo-hooks/tokens/useToken'
+
+export enum HaloSwapType {
+  SwapExactIn,
+  SwapExactOut
+}
+
+const useSwap = (tokenIn?: Token, tokenOut?: Token) => {
+  const { account } = useActiveWeb3React()
+  // const balancer = useBalancerSDK()
+  const haloAddresses = useHaloAddresses()
+  const vaultContract = useContract(haloAddresses.ammV2.vault, VaultABI)
+  const { allowTokenAmount } = useToken()
+
+  // useEffect(() => {
```

Comment:
Are we planning to still use this commented code block? @schystz

Reply:
I've actually commented out most of the amm v1 codes just to have an easy access to the old code while the new feature is still under development. Will perform cleanup here once the swap feature is finalized @tracyarciaga .